Header/graph_color.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _GRAPH_COLOR_H_
+#define _GRAPH_COLOR_H_
+
+#include <map>
+#include <stack>
+#include <locale>
+#include <vector>
+#include <string>
+#include <utility>
+#include <iostream>
+#include <queue>
+
+using std::queue;
+using std::map;
+using std::pair;
+using std::stack;
+using std::string;
+using std::vector;
+using std::cout;
+using std::cerr;
+using std::endl;
+
+namespace GraphColoring {
+    enum Algorithm {kNone,kDSATUR,kMCS,kTABUCOL,kHybrid,kLMXRLF,kHybridDSATUR};
```

Comment:
Since the algorithms are now encapsulated into their own objects, that object should have some virtual print function that will be overridden in each concrete subclass to print the correct algorithm type instead of using this enum system.